src/metrics/counter.ts

```diff
@@ -10,8 +12,10 @@ export function makeCounterProvider(
 ): Provider {
   return {
     provide: getToken(options.name),
-    useFactory(): client.Metric<string> {
+    useFactory(config: PrometheusPrefix): client.Metric<string> {
```

Comment:
I would rather you just pass through the entire `PrometheusOptions` that already exists under the `PROMETHEUS_OPTIONS` symbol. The name you chose is a misnomer because you are only passing the prefix through.